openmetadata-ui/src/main/resources/ui/playwright/utils/entityPermissionUtils.ts

```diff
@@ -0,0 +1,509 @@
+/*
+ *  Copyright 2024 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { expect, Page } from '@playwright/test';
+import { VIEW_ALL_RULE } from '../constant/permission';
+import { PolicyClass } from '../support/access-control/PoliciesClass';
+import { RolesClass } from '../support/access-control/RolesClass';
+import { ContainerClass } from '../support/entity/ContainerClass';
+import { DashboardClass } from '../support/entity/DashboardClass';
+import { DashboardDataModelClass } from '../support/entity/DashboardDataModelClass';
+import { EntityClass } from '../support/entity/EntityClass';
+import { MetricClass } from '../support/entity/MetricClass';
+import { MlModelClass } from '../support/entity/MlModelClass';
+import { PipelineClass } from '../support/entity/PipelineClass';
+import { SearchIndexClass } from '../support/entity/SearchIndexClass';
+import { TableClass } from '../support/entity/TableClass';
+import { TopicClass } from '../support/entity/TopicClass';
+import { UserClass } from '../support/user/UserClass';
+import { getApiContext, redirectToHomePage } from './common';
+
+// All operations across all entities
+const ALL_OPERATIONS = [
+  // Common operations
+  'EditDescription',
+  'EditOwners',
+  'EditTier',
+  'EditDisplayName',
+  'EditTags',
+  'EditGlossaryTerms',
+  'Delete',
+
+  // Entity specific operations
+  'ViewQueries',
+  'ViewSampleData',
+  'ViewDataProfile',
+  'ViewTests',
+  'ViewProfilerGlobalConfiguration',
+  'EditQueries',
+  'EditDataProfile',
+  'EditSampleData',
+  'EditTests',
+  'EditStatus',
```

Comment:
I see some fields missing here. Like 'EditCustomField' 'ViewUsage'. Please go through the permissions (common or specific) for each asset and cover all the possible permissions. Also the combinations of specific permission with `all` permission.

Reply:
addressed